streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -372,6 +378,24 @@ void updateRestored(final Collection<TopicPartition> restored) {
         }
     }
 
+    @Override
+     void removeTaskFromAllOldMaps(final StreamTask task, final Map<TaskId, StreamTask> newState) {
+        super.removeTaskFromAllOldMaps(task, newState);
+
+        final TaskId id = task.id();
+        final Set<TopicPartition> taskPartitions = new HashSet<>(task.partitions());
+        taskPartitions.addAll(task.changelogPartitions());
+
+        if (newState != restoring) {
```

Comment:
Not sure if this matters, but I don't see where `removeTaskFromAllOldMaps`  passing in `restoring` as the new state gets called.

Reply:
Makes sense, I was thinking that was your reasoning, but I wanted to confirm.